workflow/controller/workflowpod.go

```diff
@@ -321,6 +319,32 @@ func (woc *wfOperationCtx) createWorkflowPod(ctx context.Context, nodeName strin
 		{Name: common.EnvVarDeadline, Value: woc.getDeadline(opts).Format(time.RFC3339)},
 	}
 
+	if woc.controller.Config.Progress.Enabled == nil || *woc.controller.Config.Progress.Enabled {
+		// setup defaults for ticks if not set or set to 0 value.
+		patchPodTickDuration := woc.controller.Config.Progress.PatchPodAnnotationTickDuration
+		if patchPodTickDuration == nil {
+			patchPodTickDuration = &metav1.Duration{Duration: 1 * time.Minute}
+		}
+		readFileTickDuration := woc.controller.Config.Progress.ReadProgressFileTickDuration
+		if readFileTickDuration == nil {
+			readFileTickDuration = &metav1.Duration{Duration: 3 * time.Second}
+		}
+		envVars = append(envVars,
+			apiv1.EnvVar{
```

Comment:
I think we want to be able to disable this feature, but as a user, I'd always look or `ARGO_PROGRESS_FILE` and write it, so I think we should always set this value, even if we ignore it

Reply:
that makes sense :+1: changed it.